cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1227,6 +1231,49 @@ private boolean hasRebalanceAnnotation(KafkaRebalance kafkaRebalance) {
                 kafkaRebalance.getMetadata().getAnnotations().containsKey(ANNO_STRIMZI_IO_REBALANCE);
     }
 
+    private boolean shouldRenewRebalance(KafkaRebalance kafkaRebalance, KafkaRebalanceAnnotation rebalanceAnnotation) {
+        // check if request are made when CC is not active
+        if (kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "ConnectException".equals(condition.getReason()))) {
+            return true;
+        }
+        return  rebalanceAnnotation == KafkaRebalanceAnnotation.refresh || CruiseControlIssues.checkForMatch(kafkaRebalance.getStatus().getConditions());
```

Comment:
Can we split these into separate branches, it would improve readability

Reply:
Done